frontend/src/modules/entitydetails/components/Metadata.js

```diff
@@ -25,19 +25,58 @@ type Props = {|
     +navigateToPath: (string) => void,
 |};
 
+type State = {|
+    seeMore: boolean,
+    isNotTruncated: boolean
+|};
+
 
 /**
  * Component showing metadata of an entity.
  *
  * Shows:
  *  - the original string
  *  - a comment (if any)
+ *  - a group comment (if any)
+ *  - a resource comment (if any)
  *  - a context (if any)
  *  - a list of source files (if any)
  *  - a link to the resource
  *  - a link to the project
  */
-export default class Metadata extends React.Component<Props> {
+export default class Metadata extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            seeMore: false,
+            isNotTruncated: false
+        };
+        this.span = React.createRef();
+    }
+
+    componentDidMount() {
+        this.setState({
+            isNotTruncated: this.isEllispsisActivated()
+            });
+        }
+
+    componentDidUpdate(prevProps: Props) {
+        if (this.props.entity !== prevProps.entity) {
+            this.setState({ seeMore: false });
+        }
+    }
+
+    isEllispsisActivated = () => {
+        const span = this.span.current;
+        const element = span.querySelector('span.resource')
```

Comment:
I understand why you did that, but I wonder if we can reference that span directly somehow. Maybe @adngdb has ideas? Maybe we should not use `Property` here?

Reply:
This was the only way I was able to work out how to target this specific span as opposed to all the comments.  If @adngdb has another idea, just let me know.

Reply:
Can you not put the `ref` directly on the element you're interested in? Which one is it you're trying to reach?

Reply:
Hm... It's not in the `Property` component actually, it's here:
https://github.com/mozilla/pontoon/pull/1431/files/8cf204ec53a0de923656981f2a08a1cacfa0c9fe#diff-fff36ad131370bbb16e4d6f466368b64R154

You should be able to put it there directly?

Reply:
I'm trying to reach the resource comment, and when I placed the ref on any of the elements in `renderResourceComment` it would return null.

Reply:
Can you pass the `ref` to `Property` and inside `Property` assign it to the right element?

If you haven't done so already, I recommend reading the doc about Refs: https://reactjs.org/docs/refs-and-the-dom.html

Reply:
@adngdb I have been using the docs for Refs as well as referencing other uses of refs within the code base. 

Here is what I have found so far: 
If I pass the `ref` directly to `Property` I get the following error: 

> "Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?"

and the button does not render as the `ref` would be null.

When looking into forwarding the `ref` it appears that this is recommended for reusable component libraries and this is the only time I'm aware of that this element will need to be referenced. At any rate, my initial attempts at this were not successful, but I'm happy to keep looking into it if this is the direction needed.

Reply:
OK, I finally understood and reproduced the problem. I think you are overdoing it here, just using a fixed max length should be fine. I for one don't care too much if it doesn't look pretty (wink wink Matjaz) as long as it works. I'm also not sure why you want to hide that comment by default, and not just show all of it?

Reply:
The reason we want to hide it by default is that it's usually long and repeats for each string within the file. It was an explicit request, which we took without knowing it would cause so much trouble. :)

Reply:
@abowler2 As agreed with @adngdb, we should show the first 90 characters of the Resource comment and then add "...", followed by the "See more" button. Based on our screen resolution data, that will fit the Resource comment into one line for >95% of our users.

Reply:
I knew I ran into issues when trying to use the length as the condition for expanding the text before, but it was one of the first things I tried and I've been down many paths since, so I couldn't remember exactly what the issue was. 

I've been able to recreate it and get a little bit more insight (PR has been pushed for your review):  
In the ternary the `comment` can be seen for determining the length and for splicing and rendering the first 85 chars (90 was too wide with the button), BUT it won't render the full comment after the button is clicked. The button hides, but the comment does not change.

However, if the `comment` is less than 85 chars then it displays just fine (without the ellipsis and no button), and if I change the `comment` in the `true` value to a plain string (ie "Hello") the comment changes to 'Hello' when the button is clicked.

I'm not sure why this is happening or even how to research this issue... 🆘